libraries/Microsoft.Bot.Connector/Authentication/BotFrameworkAuthenticationFactory.cs

```diff
@@ -43,18 +44,18 @@ public static class BotFrameworkAuthenticationFactory
             HttpClient httpClient,
             ILogger logger)
         {
-            if (string.IsNullOrEmpty(channelService))
-            {
-                return new PublicCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
-            }
-            else if (channelService == GovernmentAuthenticationConstants.ChannelService)
-            {
-                return new GovernmentCloudBotFrameworkAuthentication(credentialFactory, authConfiguration, httpClient, logger);
-            }
-            else
+            if (
+                !string.IsNullOrEmpty(toChannelFromBotLoginUrl) || 
```

Comment:
`ToChannelFromBotLoginUrl` is [deprecated](https://github.com/microsoft/botbuilder-dotnet/blob/540fadc4437c88f36f8124308d5a3e779223f838/libraries/Microsoft.Bot.Connector/Authentication/AuthenticationConstants.cs#L13-L18) in public Azure.

`ToChannelFromBotLoginUrl` [isn't marked as deprecated](https://github.com/microsoft/botbuilder-dotnet/blob/540fadc4437c88f36f8124308d5a3e779223f838/libraries/Microsoft.Bot.Connector/Authentication/GovernmentAuthenticationConstants.cs#L16-L19) in Azure Government.

Should we be checking for this value in the air gapped clouds and should the Azure Government value be deprecated?

Should the SDK check for all of these values as a single entity, or can individual values (i.e. oAuthUrl, toBotFromChannelOpenIdMetadataUrl or toBotFromEmulatorOpenIdMetadataUrl) be used and/or different from what is normally associated with the ChannelService/ToChannelFromBotOAuthScope/ToBotFromChannelTokenIssuer matrix?

Reply:
perhaps @Jeffders and @willportnoy can comment on this?

Reply:
That particular constant might be deprecated, but it's likely this one is in use instead: https://github.com/microsoft/botbuilder-dotnet/blob/540fadc4437c88f36f8124308d5a3e779223f838/libraries/Microsoft.Bot.Connector/Authentication/AuthenticationConstants.cs#L25

I think there might have been some experiment to change the tenant from which we requested a token?

Reply:
I recall a work stream led by Ming for a customer that needed a more secure chat bot. This is probably that work...

@Jeffders does DL-ASE provide additional security assurances that would render the LoginUrl template less likely to be used?

If it’s still in use and we’re directing customers to this feature/configuration, do we need it in the other clouds? The SDK only supports it in Public Azure.